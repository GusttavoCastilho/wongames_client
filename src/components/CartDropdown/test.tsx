import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import itemsMock from 'components/CartList/mock'

import CartDropdown from '.'

const props = {
  items: itemsMock,
  total: 'R$ 300,00'
}

describe('<CartDropdown />', () => {
  it('should render <CartIcon /> and its badge', () => {
    const { container } = renderWithTheme(<CartDropdown {...props} />)

    expect(screen.getByLabelText(/shopping cart/i)).toBeInTheDocument()
    expect(screen.getByText(`${props.items.length}`)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render Dropdown content with cart items and total', () => {
    renderWithTheme(<CartDropdown {...props} />)

    expect(screen.getByText('R$ 300,00')).toBeInTheDocument()
    expect(screen.getByText(`${props.items[0].title}`)).toBeInTheDocument()
  })
})
