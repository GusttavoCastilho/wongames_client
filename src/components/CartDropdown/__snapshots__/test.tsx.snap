// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartDropdown /> should render <CartIcon /> and its badge 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c3 {
  width: 2.4rem;
  height: 2.4rem;
  color: #FAFAFA;
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #3CD3C1;
  color: #FAFAFA;
  font-size: 1rem;
  border-radius: 50%;
  width: 1.6rem;
  height: 1.6rem;
  position: absolute;
  top: -0.4rem;
  right: -0.4rem;
}

.c18 {
  background: linear-gradient(180deg,#ff5f5f 0%,#f052c0 50%);
  color: #FAFAFA;
  border: none;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c18:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c10 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
}

.c12 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c14 {
  font-size: 1.4rem;
  line-height: 1.4rem;
  font-weight: 600;
  color: #030517;
  margin-bottom: 0.8rem;
}

.c15 {
  color: #FAFAFA;
  padding: 0.2rem 0.8rem;
  background-color: #3CD3C1;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  font-weight: 600;
}

.c9 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c16 {
  background: #eee;
  color: #030517;
  font-weight: 600;
  font-size: 1.4rem;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17 {
  color: #F231A5;
}

.c2 {
  cursor: pointer;
  color: #FAFAFA;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 2.4rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FAFAFA;
  color: #030517;
  margin-top: 2.4rem;
  position: absolute;
  right: 0;
}

.c7::before {
  content: '';
  position: absolute;
  border-right: 1.2rem solid transparent;
  border-left: 1.2rem solid transparent;
  border-bottom: 1.2rem solid #FAFAFA;
  top: -1.2rem;
  right: 2.4rem;
}

.c1 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c1 .c6 {
  -webkit-transition: -webkit-transform 0.2s ease-in,opacity 0.3s ease-in-out;
  -webkit-transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  transition: transform 0.2s ease-in,opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-2rem);
  -ms-transform: translateY(-2rem);
  transform: translateY(-2rem);
}

.c0 .c8 {
  width: 56rem;
}

@media (min-width:768px) {
  .c10 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c12 {
    margin-right: 1.6rem;
    width: 15rem;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c14 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c16 {
    font-size: 1.6rem;
    padding: 2.4rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          aria-label="Cart Items"
          class="c4"
        >
          2
        </span>
        <svg
          aria-hidden="true"
          aria-label="Shopping Cart"
          class="c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="c6 c7"
    >
      <div
        class="c8 c9"
      >
        <div
          class="c10"
        >
          <div
            class="c11"
          >
            <div
              class="c12"
            >
              <img
                alt="Red Dead Redemption 2"
                src="https://source.unsplash.com/user/willianjusten/151x70"
              />
            </div>
            <div
              class="c13"
            >
              <h3
                class="c14"
              >
                Red Dead Redemption 2
              </h3>
              <div
                class="c15"
              >
                R$ 215,00
              </div>
            </div>
          </div>
        </div>
        <div
          class="c10"
        >
          <div
            class="c11"
          >
            <div
              class="c12"
            >
              <img
                alt="Borderlands 3"
                src="https://source.unsplash.com/user/willianjusten/151x70"
              />
            </div>
            <div
              class="c13"
            >
              <h3
                class="c14"
              >
                Borderlands 3
              </h3>
              <div
                class="c15"
              >
                R$ 215,00
              </div>
            </div>
          </div>
        </div>
        <div
          class="c16"
        >
          <span
            class="c17"
          >
            R$ 300,00
          </span>
          <a
            class="c18"
            href="/cart"
          >
            <span>
              Buy it now
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
